### ISSUE TITLE
Bug: preloading SMILES and using custom features returns TypeError

### ISSUE THREAD

**Issue by JenniferHem** [chemprop/chemprop#314]
Using chemprop from within python with custom features (via '--features_path') works only if neither the SMILES nor the models are added via arguments to chemprop.train.make_predictions().

A working example would be
```
arguments = [
    '--test_path', 'test.csv',
    '--preds_path', os.devnull,
    '--checkpoint_dir', 'chemprop_test',
    '--features_path', 'test_chemprop_descriptors.csv',
    "--no_features_scaling",
    "--no_cuda"
]

args = chemprop.args.PredictArgs().parse_args(arguments)
with contextlib.redirect_stdout(open(os.devnull, 'w')):
    preds = chemprop.train.make_predictions(args=args)
```
not working is e.g.:
```
arguments = [
    '--test_path', 'test.csv',
    '--preds_path', os.devnull,
    '--checkpoint_dir', 'chemprop_test',
    '--features_path', 'test_chemprop_descriptors.csv',
    "--no_features_scaling",
    "--no_cuda"
]

args = chemprop.args.PredictArgs().parse_args(arguments)
preds = chemprop.train.make_predictions(args=args, smiles=smiles)
```
where smiles is a numpy array of format [[SMILES],[SMILES]]
What I receive is a **TypeError**: dispatcher for __array_function__ did not return an iterable

to see the error I have compiled a notebook with examples (including trained models), the molecules look fairly strange but they were just used for testing purposes
[notebooks.tar.gz](https://github.com/chemprop/chemprop/files/9316993/notebooks.tar.gz)

System details are:
Chemprop version 1.5.1 and 1.5.2 tested with python 3.6 (unfortunately I cannot upgrade due to a legacy project), 1.5.2 was a forced install but I wanted to see if the upgrade might fix the issue here I had numpy version 1.19.5, to check if the python is the issue I also tested on python 3.8 with chemprop 1.5.1 and numpy 1.21.4.

**Comment by davidegraff** [comment]
Sorry for the delayed response. We will try and run the notebooks and get back to this. In the meantime, if you can provide the full stack trace of your error, that would be really helpful!

**Comment by kevingreenman** [comment]
Here's the stack trace from the notebook:
```
TypeError                                 Traceback (most recent call last)
<ipython-input-11-c12da5479b4e> in <module>
      9 
     10 args = chemprop.args.PredictArgs().parse_args(arguments)
---> 11 preds = chemprop.train.make_predictions(args=args, smiles=smiles)

~/Programs/anaconda3/envs/chembiosim/lib/python3.6/site-packages/chemprop/utils.py in wrap(*args, **kwargs)
    536         def wrap(*args, **kwargs) -> Any:
    537             start_time = time()
--> 538             result = func(*args, **kwargs)
    539             delta = timedelta(seconds=round(time() - start_time))
    540             info = logging.getLogger(logger_name).info if logger_name is not None else print

~/Programs/anaconda3/envs/chembiosim/lib/python3.6/site-packages/chemprop/train/make_predictions.py in make_predictions(args, smiles, model_objects, calibrator, return_invalid_smiles, return_index_dict, return_uncertainty)
    464             num_models=num_models,
    465             calibrator=calibrator,
--> 466             return_invalid_smiles=return_invalid_smiles,
    467         )
    468 

~/Programs/anaconda3/envs/chembiosim/lib/python3.6/site-packages/chemprop/train/make_predictions.py in predict_and_save(args, train_args, test_data, task_names, num_tasks, test_data_loader, full_data, full_to_valid_indices, models, scalers, num_models, calibrator, return_invalid_smiles, save_results)
    168         dropout_sampling_size=args.dropout_sampling_size,
    169         individual_ensemble_predictions=args.individual_ensemble_predictions,
--> 170         spectra_phase_mask=getattr(train_args, "spectra_phase_mask", None),
    171     )
    172 

~/Programs/anaconda3/envs/chembiosim/lib/python3.6/site-packages/chemprop/uncertainty/uncertainty_estimator.py in __init__(self, test_data, test_data_loader, models, scalers, num_models, uncertainty_method, dataset_type, loss_function, uncertainty_dropout_p, dropout_sampling_size, individual_ensemble_predictions, spectra_phase_mask)
     38             dropout_sampling_size=dropout_sampling_size,
     39             individual_ensemble_predictions=individual_ensemble_predictions,
---> 40             spectra_phase_mask=spectra_phase_mask,
     41         )
     42         self.label = self.predictor.label

~/Programs/anaconda3/envs/chembiosim/lib/python3.6/site-packages/chemprop/uncertainty/uncertainty_predictor.py in build_uncertainty_predictor(uncertainty_method, test_data, test_data_loader, models, scalers, num_models, dataset_type, loss_function, uncertainty_dropout_p, dropout_sampling_size, individual_ensemble_predictions, spectra_phase_mask)
    838             dropout_sampling_size=dropout_sampling_size,
    839             individual_ensemble_predictions=individual_ensemble_predictions,
--> 840             spectra_phase_mask=spectra_phase_mask,
    841         )
    842     return predictor

~/Programs/anaconda3/envs/chembiosim/lib/python3.6/site-packages/chemprop/uncertainty/uncertainty_predictor.py in __init__(self, test_data, test_data_loader, models, scalers, num_models, dataset_type, loss_function, uncertainty_dropout_p, dropout_sampling_size, individual_ensemble_predictions, spectra_phase_mask)
     48         self.raise_argument_errors()
     49         self.test_data_loader = test_data_loader
---> 50         self.calculate_predictions()
     51 
     52     @property

~/Programs/anaconda3/envs/chembiosim/lib/python3.6/site-packages/chemprop/uncertainty/uncertainty_predictor.py in calculate_predictions(self)
    145                 data_loader=self.test_data_loader,
    146                 scaler=scaler,
--> 147                 return_unc_parameters=False,
    148             )
    149             if self.dataset_type == "spectra":

~/Programs/anaconda3/envs/chembiosim/lib/python3.6/site-packages/chemprop/train/predict.py in predict(model, data_loader, disable_progress_bar, scaler, return_unc_parameters, dropout_prob)
     58                 atom_descriptors_batch,
     59                 atom_features_batch,
---> 60                 bond_features_batch,
     61             )
     62 

~/Programs/anaconda3/envs/chembiosim/lib/python3.6/site-packages/torch/nn/modules/module.py in _call_impl(self, *input, **kwargs)
   1100         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   1101                 or _global_forward_hooks or _global_forward_pre_hooks):
-> 1102             return forward_call(*input, **kwargs)
   1103         # Do not call functions when jit is used
   1104         full_backward_hooks, non_full_backward_hooks = [], []

~/Programs/anaconda3/envs/chembiosim/lib/python3.6/site-packages/chemprop/models/model.py in forward(self, batch, features_batch, atom_descriptors_batch, atom_features_batch, bond_features_batch)
    195 
    196         output = self.ffn(self.encoder(batch, features_batch, atom_descriptors_batch,
--> 197                                        atom_features_batch, bond_features_batch))
    198 
    199         if self.classification and not (self.training and self.no_training_normalization) and self.loss_function != 'dirichlet':

~/Programs/anaconda3/envs/chembiosim/lib/python3.6/site-packages/torch/nn/modules/module.py in _call_impl(self, *input, **kwargs)
   1100         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   1101                 or _global_forward_hooks or _global_forward_pre_hooks):
-> 1102             return forward_call(*input, **kwargs)
   1103         # Do not call functions when jit is used
   1104         full_backward_hooks, non_full_backward_hooks = [], []

~/Programs/anaconda3/envs/chembiosim/lib/python3.6/site-packages/chemprop/models/mpn.py in forward(self, batch, features_batch, atom_descriptors_batch, atom_features_batch, bond_features_batch)
    261 
    262         if self.use_input_features:
--> 263             features_batch = torch.from_numpy(np.stack(features_batch)).float().to(self.device)
    264 
    265             if self.features_only:

<__array_function__ internals> in stack(*args, **kwargs)

TypeError: dispatcher for __array_function__ did not return an iterable
```

**Comment by kevingreenman** [comment]
We are closing all bug reports that we don't have time to fix before the release of v1.7 this month, and unfortunately this is one of them. We will soon release [Chemprop v2.0](https://github.com/chemprop/chemprop/issues/517) and do not plan to maintain/support v1.x any longer. If anyone has time to open a PR that fixes this bug before the end of January 2024, we will review it and consider including it in v1.7.
